### PR TITLE
T14475

### PR DESCRIPTION
**Pull request by magcius**
https://phabricator.endlessm.com/T14475

### CODE REVIEW

File: licenseCrawler.js [hunk 1]
```diff
@@ -35,105 +33,81 @@ function prepareHtml(fileNames, copyright) {
...
-	var copyrightContents;
-	try {
-	    copyrightContents = fs.readFileSync(copyrightPath, { encoding: 'utf8' });
-	} catch (e) {
-	    console.log('Unable to read copyright file ' + copyrightPath + ' :' + e.toString());
-	    return;
-	}
+    packageNames.sort();
+    packageNames.forEach(function(dirName) {
+	const copyrightPath = CRAWL_LOCATION + '/' + dirName + '/copyright';
```
Comment (cosimoc) [inline licenseCrawler.js]: You could use `GLib.build_filenamev()`

File: licenseCrawler.js [hunk 2]
```diff
@@ -35,105 +33,81 @@ function prepareHtml(fileNames, copyright) {
...
-	var linkNames = linksHash[dirName];
-	if (!linkNames) {
-	    linkNames = [];
-	}
+        try {
+	    const [success, copyrightContents] = GLib.file_get_contents(copyrightPath);
+        } catch (e if e.matches(GLib.FileError, GLib.FileError.NOENT)) {
+            return;
+        } catch (e) {
+	    log('Unable to read copyright file ' + copyrightPath + ' :' + e.toString());
```
Comment (cosimoc) [inline licenseCrawler.js]: Use `logError()` (also indentation)

File: licenseCrawler.js [hunk 3]
```diff
@@ -35,105 +33,81 @@ function prepareHtml(fileNames, copyright) {
...
 
-	var copyrightContents;
-	try {
-	    copyrightContents = fs.readFileSync(copyrightPath, { encoding: 'utf8' });
-	} catch (e) {
-	    console.log('Unable to read copyright file ' + copyrightPath + ' :' + e.toString());
-	    return;
-	}
+    packageNames.sort();
+    packageNames.forEach(function(dirName) {
```
Comment (cosimoc) [inline licenseCrawler.js]: Nit: change the variable name here to `packageName`

File: licenseCrawler.js [hunk 4]
```diff
@@ -35,105 +33,81 @@ function prepareHtml(fileNames, copyright) {
...
+	    const [success, copyrightContents] = GLib.file_get_contents(copyrightPath);
+        } catch (e if e.matches(GLib.FileError, GLib.FileError.NOENT)) {
+            return;
+        } catch (e) {
+	    log('Unable to read copyright file ' + copyrightPath + ' :' + e.toString());
+            return;
+        }
 
-	linkNames.unshift(dirName);
+	const names = packages[dirName];
```
Comment (cosimoc) [inline licenseCrawler.js]: Nit: rename `names` to `dirNames` or `fileNames`

File: licenseApp.js [hunk 5]
```diff
@@ -1,22 +1,31 @@
...
+
+    msg.status_code = 200;
+    msg.response_headers.set_content_type('text/html', { charset: 'UTF-8' });
 
-var server = http.createServer(app);
-server.autoQuit({ timeOut: 120 });
-server.listen(app.get('port'));
-server.on('listening', function() {
-    console.log('Express server listening on port ' + app.get('port'));
+    licenseCrawler.getLicenseList(res);
```
Comment (cosimoc) [inline licenseApp.js]: Should you call `msg.response_body.complete()` after this?

Comment (magcius) [inline licenseApp.js]: https://github.com/GNOME/gjs/blob/master/examples/http-server.js doesn't -- should I?

**Pull request merged**